Hello World Example Fails to Run

**Jetty version(s)**
<!--[Jetty 9.x is now at End of Community Support](https://github.com/eclipse/jetty.project/issues/7958) -->
jetty-12

**Jetty Environment**
<!-- Applicable for jetty-12 only, choose: core, ee8, ee9, ee10 -->
core?

**Java version/vendor** `(use: java -version)`
openjdk 22.0.1 2024-04-16
OpenJDK Runtime Environment Temurin-22.0.1+8 (build 22.0.1+8)
OpenJDK 64-Bit Server VM Temurin-22.0.1+8 (build 22.0.1+8, mixed mode)

**OS type/version**
macOS 14.4.1 (23E224)

**Description**
<!-- Do not report security issues here! See [Jetty Security Reports](https://eclipse.dev/jetty/security_reports.php) -->

Following the steps of https://jetty.org/docs/jetty/12/programming-guide/maven-jetty/jetty-maven-helloworld.html#configuring-embedded-jetty-with-maven gives error:

```
[ERROR] Failed to execute goal on project hello-world: Could not resolve dependencies for project org.example:hello-world:jar:0.1-SNAPSHOT: The following artifacts could not be resolved: org.eclipse.jetty:jetty-server:jar:12.0.10-SNAPSHOT (absent): Could not find artifact org.eclipse.jetty:jetty-server:jar:12.0.10-SNAPSHOT 
```

I looked up the latest version of this artifact at https://mvnrepository.com/artifact/org.eclipse.jetty/jetty-server/12.0.10, so I tried using `12.0.10` instead of `12.0.10-SNAPSHOT`, and got a new error:

```
[ERROR]   symbol:   class HttpServletRequest
[ERROR]   location: class org.example.HelloWorld
[ERROR] /Users/LovingSalamander/development/java/maven-jetty-example/src/main/java/org/example/HelloWorld.java:[16,24] cannot find symbol
[ERROR]   symbol:   class HttpServletResponse
[ERROR]   location: class org.example.HelloWorld
[ERROR] /Users/LovingSalamander/development/java/maven-jetty-example/src/main/java/org/example/HelloWorld.java:[17,29] cannot find symbol
[ERROR]   symbol:   class ServletException
[ERROR]   location: class org.example.HelloWorld
[ERROR] /Users/LovingSalamander/development/java/maven-jetty-example/src/main/java/org/example/HelloWorld.java:[11,8] org.example.HelloWorld is not abstract and does not override abstract method handle(org.eclipse.jetty.server.Request,org.eclipse.jetty.server.Response,org.eclipse.jetty.util.Callback) in org.eclipse.jetty.server.Request.Handler
[ERROR] /Users/LovingSalamander/development/java/maven-jetty-example/src/main/java/org/example/HelloWorld.java:[20,28] cannot find symbol
[ERROR]   symbol:   variable HttpServletResponse
[ERROR]   location: class org.example.HelloWorld
[ERROR] /Users/LovingSalamander/development/java/maven-jetty-example/src/main/java/org/example/HelloWorld.java:[21,20] cannot find symbol
[ERROR]   symbol:   method setHandled(boolean)
[ERROR]   location: variable baseRequest of type org.eclipse.jetty.server.Request
```

**How to reproduce?**

Follow the steps of the tutorial

> We need to get rid of the "-SNAPSHOT" portion of that example.
> 
> Use `<jettyVersion>12.0.10</jettyVersion>` for now.

When I use that version and run the example as-is, I get the following error:

```
[ERROR]   symbol:   class HttpServletRequest
[ERROR]   location: class org.example.HelloWorld
[ERROR] /Users/LovingSalamander/development/java/maven-jetty-example/src/main/java/org/example/HelloWorld.java:[16,24] cannot find symbol
[ERROR]   symbol:   class HttpServletResponse
[ERROR]   location: class org.example.HelloWorld
[ERROR] /Users/LovingSalamander/development/java/maven-jetty-example/src/main/java/org/example/HelloWorld.java:[17,29] cannot find symbol
[ERROR]   symbol:   class ServletException
[ERROR]   location: class org.example.HelloWorld
[ERROR] /Users/LovingSalamander/development/java/maven-jetty-example/src/main/java/org/example/HelloWorld.java:[11,8] org.example.HelloWorld is not abstract and does not override abstract method handle(org.eclipse.jetty.server.Request,org.eclipse.jetty.server.Response,org.eclipse.jetty.util.Callback) in org.eclipse.jetty.server.Request.Handler
[ERROR] /Users/LovingSalamander/development/java/maven-jetty-example/src/main/java/org/example/HelloWorld.java:[20,28] cannot find symbol
[ERROR]   symbol:   variable HttpServletResponse
[ERROR]   location: class org.example.HelloWorld
[ERROR] /Users/LovingSalamander/development/java/maven-jetty-example/src/main/java/org/example/HelloWorld.java:[21,20] cannot find symbol
[ERROR]   symbol:   method setHandled(boolean)
[ERROR]   location: variable baseRequest of type org.eclipse.jetty.server.Request
```

--- a/documentation/jetty/modules/programming-guide/pages/maven-jetty/jetty-maven-helloworld.adoc
+++ b/documentation/jetty/modules/programming-guide/pages/maven-jetty/jetty-maven-helloworld.adoc
@@ -47,36 +47,45 @@ Use an editor to create the file `src/main/java/org/example/HelloWorld.java` wit
 ----
 package org.example;
 
-import jakarta.servlet.http.HttpServletRequest;
-import jakarta.servlet.http.HttpServletResponse;
-import jakarta.servlet.ServletException;
-import java.io.IOException;
-import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.io.Content;
+import org.eclipse.jetty.server.Handler;
 import org.eclipse.jetty.server.Request;
-import org.eclipse.jetty.server.handler.AbstractHandler;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.util.Callback;
 
-public class HelloWorld extends AbstractHandler
+class HelloWorldHandler extends Handler.Abstract.NonBlocking
 {
-    public void handle(String target,
-                       Request baseRequest,
-                       HttpServletRequest request,
-                       HttpServletResponse response)
-        throws IOException, ServletException
+    @Override
+    public boolean handle(Request request, Response response, Callback callback)
     {
-        response.setContentType("text/html;charset=utf-8");
-        response.setStatus(HttpServletResponse.SC_OK);
-        baseRequest.setHandled(true);
-        response.getWriter().println("<h1>Hello World</h1>");
-    }
+        response.setStatus(200);
+        response.getHeaders().put(HttpHeader.CONTENT_TYPE, "text/html; charset=UTF-8");
 
-    public static void main(String[] args) throws Exception
-    {
-        Server server = new Server(8080);
-        server.setHandler(new HelloWorld());
-
-        server.start();
-        server.join();
+        // Write a Hello World response.
+        Content.Sink.write(response, true, """
+            <!DOCTYPE html>
+            <html>
+            <head>
+              <title>Jetty Hello World Handler</title>
+            </head>
+            <body>
+              <p>Hello World</p>
+            </body>
+            </html>
+            """, callback);
+        return true;
     }
+}
+
+Server server = new Server();
+Connector connector = new ServerConnector(server);
+server.addConnector(connector);
+
+// Set the Hello World Handler.
+server.setHandler(new HelloWorldHandler());
+
+server.start();
 }
 ----
 
@@ -116,7 +125,6 @@ Use an editor to create the file `pom.xml` in the `JettyMavenHelloWorld` directo
       <plugin>
         <groupId>org.codehaus.mojo</groupId>
         <artifactId>exec-maven-plugin</artifactId>
-        <version>1.1</version>
         <executions>
           <execution><goals><goal>java</goal></goals></execution>
         </executions>

